lib/msal-core/src/Utils.ts

```diff
@@ -724,4 +726,18 @@ export class Utils {
 
   //#endregion
 
+  //#region Response Helpers
+
+  static setResponseIdToken(response: AuthResponse, idToken: IdToken) : void {
+    response.idToken = idToken;
```

Comment:
same thing here, lets have our functions be pure so the same input always produces the same output

this could be withResponseIdToken(response, idToken)

Reply:
the name is still confusing to me.. I don't see a verb

Reply:
I am not going to debate naming here. I will call this setResponseIdToken and we can discuss naming later if it is critical.

Reply:
to me the naming is not the important part, it's that we have functions with side effects

```js
const obj = { a: 'a' };

sideEffectFunction(objectPassed) : void {
    objectPassed.b = 'b';
}

sideEffectFunction(obj);

```
Side effects are hard to reason about, and it makes it hard to tell the state of an application
my suggestion is this

```js
const myObj = { a: 'a' };

// returns new object, old one is never altered
nonSideEffectFunction(passedObject): object {
   return { ...passedObject, b: 'b' };
}

const myNewDecoratedObj = nonSideEffectFunction(myObj);

console.log(myObj); // { a: 'a' }
console.log(myNewDecoratedObj); // { a: 'a', b:'b' } 
```

Reply:
I have changed it to return an object.

Reply:
+1

Reply:
Did it get pushed? I don't see it

Reply:
name could still be clearer, that its more of a responseWithThisThing, but I am fine with it, its internal fn if we want to change in the future.